### PR TITLE
gh-103109: Add documentation for `ignore_warnings`

### PR DESCRIPTION
**Pull request by CharlieZhao95**
* Add documentation for `ignore_warnings`
* Improve the comment in function `ignore_warnings`

* Issue: gh-103109

### CODE REVIEW

File: Doc/library/test.rst
Comment (CAM-Gerlach) [inline Doc/library/test.rst]:
```suggestion
   specific known warnings in a test case, use code like the following::
```

Fix several grammar errors and phrasing issues

**Review by CAM-Gerlach — COMMENTED** [review]
**Standard reminder: You can directly apply the suggestion below by simply clicking <kbd>Commit suggestion</kbd>, entering a descriptive commit message and then clicking <kbd>Commit changes</kbd>**

One fix for some textual errors/issues, otherwise LGTM - thanks @CharlieZhao95 !

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @CharlieZhao95 !

**Review by erlend-aasland — CHANGES_REQUESTED** [review]
Please add a `versionadded:: 3.8`. ~Also, I think there's some relevant information in the docstring that should be included in the docs, namely that this decorator should be used with care since tools like `git blame` may become *less* useful with warnings suppressed.~ Forget that last part; too little coffee for me! ☕ ☕ ☕ 

Also note that there's a targeted variant called `ignore_warnings_from` (that's also undocumented[^1]).

[^1]: Note that a lot of the test support functions are undocumented, some deliberately so, simply because they're mostly for internal use.

**Comment by erlend-aasland** [comment]
Oh BTW, would you mind reflowing this according to SemBr (within 79 chars line width)? If not, I'll do that before merging.

**Comment by CharlieZhao95** [comment]
> would you mind reflowing this according to SemBr (within 79 chars line width)? If not, I'll do that before merging.

Oh! Thanks for your suggestion. I love these nits! This doesn't look too hard, I can try to get it done, and I'll try to follow the SemBr spec going forward. Improvements are always good! 😃

File: Doc/library/test.rst
Comment (CAM-Gerlach) [inline Doc/library/test.rst:1704]:
```suggestion
          # do something

   .. versionadded:: 3.8          
```

Add versionadded per @erlend-aasland 's comment (sorry, though it was added in 3.10 like the rest, but in fact the notation above was due to them being moved to a new submodule instead).

**Comment by CharlieZhao95** [comment]
> Also note that there's a targeted variant called `ignore_warnings_from` (that's also undocumented[1](https://github.com/python/cpython/pull/103110#user-content-fn-1-bab5a61b2aebdb9fbdb79125639b696b)).

BTW, I didn't find the code that matches `ignore_warnings_from`, please let me know if it is my problem. :)

**Comment by CAM-Gerlach** [comment]
> BTW, I didn't find the code that matches `ignore_warnings_from`, please let me know if it is my problem. :)

FWIW I couldn't find it either, grepping the codebase for that string and also searching the repo, issues, commits, PRs, etc. on GitHub (except for this one, of course).

**Comment by erlend-aasland** [comment]
Sorry, that was a typo on my side :) It's `ignore_deprecations_from`:

https://github.com/python/cpython/blob/c1e71ce56fdb3eab62ad3190d09130f800e54610/Lib/test/support/__init__.py#L2217-L2225

File: Doc/library/test.rst
Comment (erlend-aasland) [inline Doc/library/test.rst]: Sorry to pick on this further, but I think this may be too verbose for the `test.support` docs, as they are primarily written for CPython core devs and contributors.

I'd prefer a simpler variant, kind of like this:

```suggestion
   :exc:`Warning`.
   Example use::
```

Alternative:

```suggestion
   the :exc:`Warning` class.
   Example use::
```

@CAM-Gerlach, what do you think?

Comment (CAM-Gerlach) [reply Doc/library/test.rst]: It could definitely say the same thing in a lot fewer words, though considering this whole page is specifically for internal use, I wouldn't cut out _that_ many relevant details.—in particular, the  Also, looking at it more, I spotted some other things that can be fixed/improved. I've made another suggestion covering the full block that does both (and uses SemBr).

Comment (CAM-Gerlach) [inline Doc/library/test.rst]:
```suggestion
   Suppress warnings that are instances of *category*,
   which must be :exc:`Warning` or a subclass.
   Roughly equivalent to :func:`warnings.catch_warnings`
   with :meth:`warnings.simplefilter('ignore', category=category) <warnings.simplefilter>`.
   For example::
```

* Rewrite more clearly and concisely
* Clarify explicit what *category* does
* Avoid implying that *category* must be a _strict_ subclass of `Warning`
* State what it is equivalent to more explicitly.
* Fix some minor grammar/phrasing issues I missed the first time

Comment (erlend-aasland) [reply Doc/library/test.rst]: Looks good to me. @CharlieZhao95 can you apply CAM's suggestion?

Comment (CharlieZhao95) [reply Doc/library/test.rst]: > Looks good to me. @CharlieZhao95 can you apply CAM's suggestion?

Sure. Thanks to everyone for the careful review and suggestions 😄

Comment (CAM-Gerlach) [reply Doc/library/test.rst]: Thanks for your contribution!

Comment (CAM-Gerlach) [reply Doc/library/test.rst]: Looks like the agreed suggestion was applied, so marking this as resolved, thanks.

**Review by CAM-Gerlach — APPROVED** [review]
LGTM again, thanks @CharlieZhao95

**Review by erlend-aasland — APPROVED** [review]

**Pull request merged**
